### ISSUE TITLE
Wrong size for integers

### ISSUE THREAD

**Issue by electricentity** [usnistgov/REFPROP-wrappers#380]
### Description

integers are treated as longs, which on systems that have 64 bit longs results in negative values being dramatically off.

### Steps to Reproduce

Checking error values on a 64bit Linux distribution results in negative values (like -117) which should be a warning being reported as a positive 64 bit value (like 4294967179)

### Versions

Self compiled linux distro from REFPROP-cmake
Ubuntu 18
Python ctREFPROP library

### Additional Information

Looking at the REFPROP.h file the types for integers are explicitly set as "int" but the ctypes wrapper utilizes `c_long` which will use whatever the native size of `long` is. This is acceptable for a Windows system or a 32-bit system but 64-bit Linux has longs being 64-bits which means the 32-bit 2's compliment value is severely misinterpreted.

It seems like this should be fairly easy to fix by changing the `build_wrapper.py/gen_ctypes_wrappers` function to use `ct.c_int32`

I may also have messed something up in all of the building of the Linux shared library but looking at the `generate_header.py` file it seems like it can't utilize anything other than an int for integer values so it seems like the ctypes wrapper needs to be modified to be more strict in the size of the outputs.

**Comment by ianhbell** [comment]
We've been testing/running this wrapper on all platforms for quite a few years now and I myself use 64-bit python on linux.  We tested this extensively before, and this seemed like the right solution.  See also: https://github.com/CoolProp/REFPROP-headers#notes-on-integers

**Comment by ianhbell** [comment]
Note that values of -117 are correct in some cases, in this case for a mixture model with estimated interaction parameters.  Can you show a case where you get a different result from the same input on windows and linux? That would be most interesting and worth further investigation.

**Comment by electricentity** [comment]
First, thanks for getting back to me so quick. I'm kinda hoping you are no longer in Boulder given 4 AM seems like a crazy time to be responding to issue tickets. 

Hmmm... given you have been using 64-bit python on linux maybe I did just mess something up with my compilation. I'll mess around with it a bit today.

I agree with the CooProp use of int, my confusion mainly lies with the ctREFPROP use of `c_long` to interact with the `int` inputs/outputs. [ctypes](https://docs.python.org/3/library/ctypes.html#ctypes.c_int) does not strictly treat `c_int` and `c_long` as the same if the system has them as different sizes. 

This seems to explain the behavior I am getting of expected warning values (like `-117`) being interpreted as `4294967179`, which is how a 32-bit signed integer representation of `-117` would be interpreted if the 4 bytes representing that int were put into the 8 bytes of a 64-bit integer.

The problem I am seeing is actually in some unit tests that I have been using on Windows with the standard DLLs and then was using to verify that the linux shared library was working properly with the rest of my code on a linux system. The test expects to see a warning (with an `ierr` value of `-117`) but treats it as an error because it gets a value of `4294967179`. For some reason my error handling code assumes that any `ierr > 100` is an error and anything else is a warning but while typing this I was unable to actually find the REFPROP documentation to justify that check although it seems to hold up. When I pass the `ierr` back into `ERRMSGdll` I end up getting the correct `herr` as the 8 byte value is passed into a 4 byte value again. However given this seems like it would be an issue for any int parameter that is being checked outside of the shared library I would like to figure out what is causing it to prevent it from presenting itself unexpectedly somewhere else.

I am curious what your thoughts are with respect to using `c_long` vs `c_int` given the stance that you took on integers for the CoolProp headers.

**Comment by ianhbell** [comment]
I live on East Coast, so 6 am is slightly more socially acceptable. :)

What bitness do you use for your linux tests? I'm surprised I haven't run into this before.

I agree that int type sizes is one of the most challenging things to get right in interfacing with FORTRAN/C/C++.  Please see this note in the ctypes docs: https://docs.python.org/3/library/ctypes.html#ctypes-tutorial . 

The ierr codes greater than 100 are indeed fatal errors, values  > 0 and < 100 are bad, but not fatal, errors.

Can you give me a single set of inputs that cause this error so I can do some digging?

**Comment by ianhbell** [comment]
I think if we were to switch everywhere to ``c_int`` in the ctypes interface, that would fix it, no?

**Comment by ianhbell** [comment]
Wow. You're totally right...  This gives -117 on windows, and 4294967179 on 64-bit linux. 

```python
import os
from ctREFPROP.ctREFPROP import REFPROPFunctionLibrary

root = os.environ['RPPREFIX']
RP = REFPROPFunctionLibrary(root)
RP.SETPATHdll(root)

print(RP.SETFLUIDSdll('R1234ZEE*ETHANE'))
```

**Comment by ianhbell** [comment]
I'll fix this today!

**Comment by ianhbell** [comment]
Confirmed at least in the ``SETFLUIDSdll`` case that switching to c_int fixes the problem and I get -117 on linux again.

**Comment by ianhbell** [comment]
Fixed: https://github.com/usnistgov/REFPROP-wrappers/commit/433ffd26436c6b24ed4149637c0723a9e418a9d2

Released version 0.10, which should not have this problem.

**Comment by electricentity** [comment]
Sweet! The case I have that was causing this was using a bunch of random fluids and mixtures that I honestly don't know the origin of so I was going to have to try to find another case that tripped that warning.

Thanks for jumping on this. Glad that I wasn't completely crazy. 

Different note: where is the documentation for what is a fatal error vs bad? I tried searching through most of the sources of documentation I am aware of and didn't come up with anything. Presumably I found something a while back given I have that check in my code but would kinda like to point to the justification for that check so that in the future I am not relying on my past self to have done something correctly without any way of proving it is correct.
